src/devices/cddasongloader.cpp

```diff
@@ -47,18 +69,86 @@ QUrl CddaSongLoader::GetUrlFromTrack(int track_number) const {
 }
 
 void CddaSongLoader::LoadSongs() {
-  QtConcurrent::run(this, &CddaSongLoader::LoadSongsFromCdda);
+  if (!disc_.has_been_read || HasChanged()) {
+    if (!loading_future_.isRunning()) {
```

Comment:
There's isStarted and isRunning. I assumed that isStarted() meant that QtConcurrent::run had been called and that isRunning() meant that the thread entry point had been called. But I may be wrong about that. The QFuture documentation isn't terribly helpful.

Reply:
Briefly looking at the source of QFuture didn't help me to figure out the difference either. I tried making that condition  `!(loading_future_.isRunning() || loading_future_.isStarted())` to cover both but found that that prevents loading completely. Apparently it holds that `loading_future_.isStarted() == true` after initialisation of `loading_future_` in the class constructor, which seems really odd to me...